test/intl402/DurationFormat/constructor-options-order.js

```diff
@@ -0,0 +1,41 @@
+// Copyright 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.DurationFormat
+description: Checks the order of operations on the options argument to the DurationFormat constructor.
+info: |
+    Intl.DurationFormat ( [ locales [ , options ] ] )
+    (...)
+    5. Let matcher be ? GetOption(options, "localeMatcher", "string", « "lookup", "best fit" », "best fit").
+    6. Let numberingSystem be ? GetOption(options, "numberingSystem", "string", undefined, undefined).
+    13. Let style be ? GetOption(options, "style", "string", « "long", "short", "narrow", "digital" », "long").
+includes: [compareArray.js]
+features: [Intl.DurationFormat]
+---*/
+
+var actual = [];
+
+const options = {
+  get localeMatcher() {
+    actual.push("localeMatcher");
+    return undefined;
+  },
+  get numberingSystem() {
+    actual.push("numberingSystem");
+    return undefined;
+  },
+  get style() {
+    actual.push("style");
+    return undefined;
+  },
+};
+
+const expected = [
+  "localeMatcher",
+  "numberingSystem",
+  "style"
+];
```

Comment:
Optional: There is a helper for testing the order of operations of property gets, although maybe it should get a more general place than TemporalHelpers if we're going to use it here as well.

```suggestion
const options = {};
TemporalHelpers.observeProperty(actual, options, "localeMatcher", undefined);
TemporalHelpers.observeProperty(actual, options, "numberingSystem", undefined);
TemporalHelpers.observeProperty(actual, options, "style", undefined);

const expected = [
  "get localeMatcher",
  "get numberingSystem",
  "get style"
];
```

Reply:
That's nice, I'm ok with using it, no strong preference at moment, only  the "TemporalHelpers" name, would be really nice to have it as generic utils functions and documented somewhere else cause they're really handy